File: examples/example-jpa-spring/src/test/java/example/repositories/PetRepositoryTest.java
package example.repositories;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

import javax.inject.Inject;

import org.junit.jupiter.api.Test;

import example.domain.Pet;
import io.micronaut.context.BeanContext;
import io.micronaut.data.annotation.Query;
import io.micronaut.test.annotation.MicronautTest;

@MicronautTest
public class PetRepositoryTest {

    @Inject
    PetRepository petRepository;


    @Inject
    BeanContext beanContext;

    @Test
    void testQuery() {
        String query = beanContext.getBeanDefinition(PetRepository.class)
                .getRequiredMethod("findByName", String.class)
                .getAnnotationMetadata().stringValue(Query.class).get();

        System.out.println("query = " + query);
    }


    @Test
    void testRetrievePetAndOwner() {
        Pet dino = petRepository.findByName("Dino").orElse(null);
        assertNotNull(dino);
        assertEquals("Dino", dino.getName());
        assertEquals("Fred", dino.getOwner().getName());
    }
}

File: examples/example-jdbc-kotlin/src/test/kotlin/example/repositories/PetRepositoryTest.kt
package example.repositories

import io.micronaut.test.annotation.MicronautTest
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test

@MicronautTest
class PetRepositoryTest(private val petRepository: PetRepository) {

    @Test
    fun testRetrievePetAndOwner() {
        val dino = petRepository.findByName("Dino").orElse(null)
        assertNotNull(dino)
        assertEquals("Dino", dino.name)
        assertEquals("Fred", dino.owner.name)
    }
}
